newlib/libc/include/sys/cdefs.h

```diff
@@ -536,12 +536,16 @@
 #define	__printf0like(fmtarg, firstvararg)
 #endif
 
-#if defined(__GNUC__)
 #if defined(_HAVE_ALIAS_ATTRIBUTE)
 #define __strong_reference_alias(n) #n
 #define	__strong_reference(sym,aliassym)	\
     extern __typeof (sym) aliassym __attribute__ ((__alias__ (__strong_reference_alias(sym))))
 #endif
+#if defined(_HAVE_WEAK_ATTRIBUTE)
+#define __weak_reference_alias(n) #n
+#define	__weak_reference(sym,aliassym)	\
+    extern __typeof (sym) aliassym __attribute__ ((__weak__, __alias__ (__weak_reference_alias(sym))))
```

Comment:
Is the extra macro above needed?

```suggestion
    extern __typeof (sym) aliassym __attribute__ ((__weak__, __alias__ (__XSTRING(sym))))
```
Or __STRING if you want to avoid expansion.

Reply:
Thanks. I've applied this fix to the series and added a patch to apply it to __strong_reference too.